### PR TITLE
Move `_CCCL_COMPILER_GCC` to the new macro

### PR DESCRIPTION
**Pull request by davebayer**
Use the new macro for GCC. This PR should be merged after #2849.

### CODE REVIEW

**Comment by miscco** [comment]
pre-commit.ci autofix

File: libcudacxx/include/cuda/std/__memory/uninitialized_algorithms.h [hunk 1]
```diff
@@ -639,7 +639,7 @@ template <class _Type>
 struct __allocator_has_trivial_move_construct<allocator<_Type>, _Type> : true_type
 {};
 
-#ifndef _CCCL_COMPILER_GCC
+#if _CCCL_COMPILER(GCC)
```
Comment (miscco) [inline libcudacxx/include/cuda/std/__memory/uninitialized_algorithms.h:642]: This is inverted
```suggestion
#if !_CCCL_COMPILER(GCC)
```

File: libcudacxx/include/cuda/std/__memory/uninitialized_algorithms.h [hunk 2]
```diff
@@ -665,7 +665,7 @@ __uninitialized_allocator_move_if_noexcept(_Alloc&, _Iter1 __first1, _Iter1 __la
     return _CUDA_VSTD::move(__first1, __last1, __first2);
   }
 }
-#endif // _CCCL_COMPILER_GCC
+#endif // _CCCL_COMPILER(GCC)
```
Comment (miscco) [inline libcudacxx/include/cuda/std/__memory/uninitialized_algorithms.h:668]:
```suggestion
#endif // !_CCCL_COMPILER(GCC)
```

File: thrust/testing/transform.cu [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 // There is a unfortunate miscompilation of the gcc-11 vectorizer leading to OOB writes
 // Adding this attribute suffices that this miscompilation does not appear anymore
-#if defined(_CCCL_COMPILER_GCC) && __GNUC__ >= 11
+#if _CCCL_COMPILER(GCC, >=, 11)
 #  define THRUST_DISABLE_BROKEN_GCC_VECTORIZER __attribute__((optimize("no-tree-vectorize")))
```
Comment (miscco) [inline thrust/testing/transform.cu:14]: Unrelated we should drop that workaround now that @ericniebler has found the root cause

**Review by miscco — CHANGES_REQUESTED** [review]

**Comment by davebayer** [comment]
pre-commit.ci autofix

**Comment by miscco** [comment]
Thanks a lot for improving our infrastructure

**Pull request merged**